### ISSUE TITLE
Specify a column on insertOrUpdate

### ISSUE THREAD

**Issue by vansimonsen** [kotlin-orm/ktorm#148]
If you want to write into a db using `insertOrUpdate` strategy, you cannot specify column to be used at `on conflict` and is assuming that is the primary key.

That isn't useful on tables with unique columns but no primary keys (for example with postgres with timescaledb extension, your primary key is an index and not and specific column)

Could I specify a column to be used to looking for a conflict or set an `ON CONSTRAINT` clause?

**Comment by vincentlauvlwj** [comment]
Thank you. This suggestion is really useful. I will implement this in a future release. The syntax might look like: 

```kotlin
database.insertOrUpdate(Employees) {
    it.id to 1
    it.name to "vince"
    it.job to "engineer"
    it.salary to 1000
    it.hireDate to LocalDate.now()
    it.departmentId to 1

    onConflict(it.id) {
        it.salary to it.salary + 900
    }
}
```

**Comment by vansimonsen** [comment]
I worked on a temporary solution at my project, it looks as:

```
class UpsertStatementBuilder(
    assignments: MutableList<ColumnAssignmentExpression<*>>
) : AssignmentsBuilder(assignments) {

    internal val updateAssignments = ArrayList<ColumnAssignmentExpression<*>>()

    /**
     * Specify the update assignments while any key/index conflict exists.
     */
    fun onDuplicateConstraint(block: AssignmentsBuilder.() -> Unit) {
        val assignments = ArrayList<ColumnAssignmentExpression<*>>()
        AssignmentsBuilder(assignments).apply(block)
        updateAssignments += assignments
    }
}

fun <T : BaseTable<*>> Database.upsert(table: T, indexes: List<ColumnExpression<*>>? = null, block: UpsertStatementBuilder.(T) -> Unit): Int {
    val assignments = ArrayList<ColumnAssignmentExpression<*>>()
    val builder = UpsertStatementBuilder(assignments).apply { block(table) }

    if (indexes == null) {
        table.primaryKey ?: error("Table ${table.tableName} doesn't have a primary key.")
    }
    val conflictTarget = indexes ?: listOf(table.primaryKey!!.asExpression())
    val expression = InsertOrUpdateExpression(
        table = table.asExpression(),
        assignments = assignments,
        conflictTarget = conflictTarget,
        updateAssignments = builder.updateAssignments
    )

    return executeUpdate(expression)
}
```

**Comment by vincentlauvlwj** [comment]
@afezeria implemented this feature. #181

**Comment by vincentlauvlwj** [comment]
Included in Ktorm 3.1.0 and its released.
